src/commons/services/utils.js

```diff
@@ -108,6 +108,35 @@
             return clusterName;
         }
 
+        vm.getVolumeDetails = function(clusterId) {
+            volumeList = [];
+            clusterData = vm.clusterData.clusters;
+            len = clusterData.length;
+            for ( i = 0; i < len; i++ ) {
+                clusterObj = clusterData[i];
+                for( key in clusterObj ) {
```

Comment:
I have one suggestion that as we know that key[0] in clusterObj will be used for fetching data, so can we just use key[0] instead of iterating over the keys of clusterObj. Because in this case we will have n2 complexity instead n3.